packages/central-server/src/apiV2/taskComments/assertTaskCommentPermissions.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2024 Beyond Essential Systems Pty Ltd
+ */
+
+import { hasBESAdminAccess } from '../../permissions';
+import { createTaskDBFilter } from '../tasks/assertTaskPermissions';
+
+export const createTaskCommentDBFilter = async (accessPolicy, models, criteria, options) => {
+  if (hasBESAdminAccess(accessPolicy)) {
+    return { dbConditions: criteria, dbOptions: options };
+  }
+  const { dbConditions } = await createTaskDBFilter(accessPolicy, models, criteria, options);
```

Comment:
```suggestion
  const { dbConditions } = await createTaskDBFilter(accessPolicy, models);
```
Just eyeballing this one, but I think we'd need to do something like this? Otherwise we'd be filtering the `task` table by fields from the `task_comment` table. Possibly we'd want to pass through the `project_id` but not sure if it's necessary